### PR TITLE
Use FhirRouter for FHIR operations

### PR DESCRIPTION
**Pull request by codyebberson**
Before: Most FHIR operation routes were handled by the Express router, not by the `FhirRouter`

That wasn't inherently bad, but it meant that those FHIR operations were not accessible via FHIR batch operations.

After:  Most FHIR operation routes are now handled by `FhirRouter`, and therefore are available via FHIR batch.

This paves the way in the future for projects to be able to configure Medplum Bots as FHIR operation handlers...

Note I said "most", as some operations are currently not compatible with the `FhirRouter` API:

* `$csv` - exports a CSV representation of search results, and therefore sets content-type, etc
* `Bot/$execute` - Bot execution can return HL7, DICOM, etc
* `Agent/$push` - Agents can return HL7, DICOM, etc

The diff is large, but it's actually pretty straightforward and procedural changes.

Two of our older FHIR operations predate @mattwiller's FHIR `Parameter` utils, so I went ahead and ported those over:

* `Measure/$evaluate-measure`
* `PlanDefinition/$apply`

In both cases, the transition was pretty straightforward, just different error messages.

### CODE REVIEW

File: dangerfile.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ if (packageChanged && !lockfileChanged) {
 const modifiedFiles = danger.git.modified_files.filter((path) => /\/src\/.+\.tsx?/.exec(path));
 
 // Check for console.log statements
-const statements = ['console.debug', 'console.log', 'console.warn', 'describe.only', 'test.only'];
```
Comment (mattwiller) [inline dangerfile.ts:19]: Was this removal intentional?

File: packages/server/src/fhir/operations/README.md [hunk 2]
```diff
@@ -58,13 +58,13 @@ interface ProjectInitParameters {
   ownerEmail?: string;
 }
 
-export async function projectInitHandler(req: Request, res: Response): Promise<void> {
+export async function projectInitHandler(req: FhirRequest): Promise<FhirResponse> {
```
Comment (mattwiller) [inline packages/server/src/fhir/operations/README.md:61]: Thanks for updating the docs as well!

File: packages/server/src/fhir/operations/evaluatemeasure.ts [hunk 3]
```diff
@@ -1,18 +1,41 @@
...
+    { name: 'periodStart', use: 'in', min: 1, max: '1', type: 'date' },
+    { name: 'periodEnd', use: 'in', min: 1, max: '1', type: 'date' },
+    { name: 'measure', use: 'in', min: 0, max: '1', type: 'Reference' },
+    { name: 'reportType', use: 'in', min: 0, max: '1', type: 'code' },
+    { name: 'subject', use: 'in', min: 0, max: '1', type: 'Reference' },
+    { name: 'practitioner', use: 'in', min: 0, max: '1', type: 'Reference' },
+    { name: 'lastReceivedOn', use: 'in', min: 0, max: '1', type: 'dateTime' },
+    { name: 'return', use: 'out', min: 1, max: '1', type: 'MeasureReport' },
+  ],
+};
```
Comment (mattwiller) [inline packages/server/src/fhir/operations/evaluatemeasure.ts:38]: Why not use the [actual FHIR spec definition](http://hl7.org/fhir/R4/measure-operation-evaluate-measure.html) of this one?

```ts
const operation = getOperationDefinition('Measure', 'evaluate-measure');
```

File: packages/server/src/fhir/operations/plandefinitionapply.ts [hunk 4]
```diff
@@ -1,41 +1,49 @@
...
+    { name: 'practitioner', use: 'in', min: 0, max: '1', type: 'string', searchType: 'reference' },
+    { name: 'organization', use: 'in', min: 0, max: '1', type: 'string', searchType: 'reference' },
+    { name: 'userType', use: 'in', min: 0, max: '1', type: 'CodeableConcept' },
+    { name: 'userLanguage', use: 'in', min: 0, max: '1', type: 'CodeableConcept' },
+    { name: 'userTaskContext', use: 'in', min: 0, max: '1', type: 'CodeableConcept' },
+    { name: 'setting', use: 'in', min: 0, max: '1', type: 'CodeableConcept' },
+    { name: 'settingContext', use: 'in', min: 0, max: '1', type: 'CodeableConcept' },
+    { name: 'return', use: 'out', min: 1, max: '1', type: 'CarePlan' },
+  ],
+};
```
Comment (mattwiller) [inline packages/server/src/fhir/operations/plandefinitionapply.ts:43]:
```suggestion
const operation = getOperationDefinition('PlanDefinition', 'apply');
```

File: packages/server/src/fhir/operations/utils/parameters.ts [hunk 5]
```diff
@@ -223,13 +187,8 @@ export async function sendOutputParameters(
     }
   }
 
-  try {
-    validateResource(response);
-    res.status(getStatus(outcome)).json(response);
-  } catch (err: any) {
-    getLogger().error('Malformed operation output Parameters', { error: err.toString() });
```
Comment (mattwiller) [inline packages/server/src/fhir/operations/utils/parameters.ts:230]: Up to you whether we want to keep this log; it was intended to help catch server issues

**Review by mattwiller — APPROVED** [review]

File: dangerfile.ts (hunk 1 above)
Comment (codyebberson) [reply dangerfile.ts:19]: Yes...  We just added this yesterday.  I was on the fence about `console.log` and all, and this PR triggered a bunch of false positives.

The "recipe" that we're currently using scans the entire file if you touch a single line.  So, in this case, I touched `deploy.ts` which has valid use of `console.log()` (inside the bot), and no way to escape it / ignore it.

Ideally, (1) Danger.js would only look at lines changed and (2) Danger.js would provide an escape hatch for "ignore".

That's low pri future work though.

File: packages/server/src/fhir/operations/evaluatemeasure.ts (hunk 3 above)
Comment (codyebberson) [reply packages/server/src/fhir/operations/evaluatemeasure.ts:38]: Oh, nice.

File: packages/server/src/fhir/operations/README.md (hunk 2 above)
Comment (codyebberson) [reply packages/server/src/fhir/operations/README.md:61]: const operation = getOperationDefinition('Measure', 'evaluate-measure');

Comment (codyebberson) [reply packages/server/src/fhir/operations/README.md:61]: "Replace in files" found it 🤣
